tutorials/ExternalWindow.EventEmitter.md

```diff
@@ -0,0 +1,399 @@
+When an instance of `fin.ExternalWindow` is created, it inherits an `EventEmitter` with the below methods so that it is possible to listen to OpenFin events. The below methods are asynchronous as they must cross process boundaries and setup the listener in the external window.  When the `EventEmitter` receives an event from the browser process and emits on the renderer, all of the functions attached to that specific event are called synchronously.  Any values returned by the called listeners are ignored and will be discarded.
+It is important to keep in mind that when an ordinary listener function is called, the standard `this` keyword is intentionally set to reference the `EventEmitter` instance to which the listener is attached.  It is possible to use ES6 Arrow Functions as listeners, however, when doing so, the `this` keyword will no longer reference the `EventEmitter` instance.
+
+#### {@link Window#addListener addListener(event, listener)}
+Adds a listener to the end of the listeners array for the specified event.
+```js
+const externalWindows = await fin.System.getAllExternalWindows();
+const notepadWin = externalWindows.find(win => win.name === 'Notepad'); // assuming only one instance of notepad is open
+
+notepadWin.addListener("bounds-changed", function(event) {
```

Comment:
Suggestion: since this is new (docs), maybe we should `await` `addListener`, `on`, `once`, etc.. ?

Reply:
I'm not sure this applies here - the functions passed to those calls are the listeners, and the calls themselves are synchronous afaik... am I missing something?

Reply:
shouldn't the actual `addListener` function be async? https://github.com/HadoukenIO/js-adapter/blob/c6083fdf213bc29310ecb6a876c6c9968ff10290/src/api/base.ts#L124

Reply:
It is

Reply:
changed.